src/py/flwr_example/quickstart_pytorch/mnist.py

```diff
@@ -0,0 +1,434 @@
+        scheduler.step()
+    return num_examples_train
+
+
+def test(
+    model: torch.nn.ModuleList,
+    test_loader: torch.utils.data.DataLoader,
+    device: torch.device = torch.device("cpu"),
+) -> Tuple[int, float, float]:
+    """Test routine 'Basic MNIST Example'
+
+    Parameters
+    ----------
+    model: torch.nn.ModuleList :
+        Neural network model used in this example.
+        
+    test_loader: torch.utils.data.DataLoader :
+        DataLoader used in test.
+        
+    device: torch.device :
+         (Default value = torch.device("cpu"))
+         Device where the network will be tested within a client.
+
+    Returns
+    -------
+        Tuple containing the total number of test samples, the test_loss, and the accuracy evaluated on the test set.
+
+    """
+    model.eval()
+    test_loss: float = 0
+    correct: int = 0
+    num_test_samples: int = 0
+    with torch.no_grad():
+        for data, target in test_loader:
+            data, target = data.to(device), target.to(device)
+            num_test_samples += len(data)
+            output = model(data)
+            test_loss += F.nll_loss(
+                output, target, reduction="sum"
+            ).item()  # sum up batch loss
+            pred = output.argmax(
+                dim=1, keepdim=True
+            )  # get the index of the max log-probability
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= num_test_samples
+
+    return (num_test_samples, test_loss, correct / num_test_samples)
+
+
+class PytorchMNISTClient(fl.client.Client):
+    """Flower client implementing MNIST handwritten classification using PyTorch."""
+
+    def __init__(
+        self,
+        cid: int,
+        train_loader: datasets,
+        test_loader: datasets,
+        epochs: int,
+        device: torch.device = torch.device("cpu"),
+    ) -> None:
+        self.model = MNISTNet().to(device)
+        self.cid = cid
+        self.train_loader = train_loader
+        self.test_loader = test_loader
+        self.device = device
+        self.epochs = epochs
+
+    def get_weights(self) -> fl.common.Weights:
+        """Get model weights as a list of NumPy ndarrays."""
+        return [val.cpu().numpy() for _, val in self.model.state_dict().items()]
+
+    def set_weights(self, weights: fl.common.Weights) -> None:
+        """Set model weights from a list of NumPy ndarrays.
+
+        Parameters
+        ----------
+        weights: fl.common.Weights 
+            Weights received by the server and set to local model
+
+
+        Returns
+        -------
+
+        """
+        state_dict = OrderedDict(
+            {
+                k: torch.Tensor(v)
+                for k, v in zip(self.model.state_dict().keys(), weights)
+            }
+        )
+        self.model.load_state_dict(state_dict, strict=True)
+
+    def get_parameters(self) -> fl.common.ParametersRes:
+        """Encapsulates the weight into Flower Parameters """
```

Comment:
Typo: `weights`

Reply:
Solved in b1559aeb7d39137b1bdc8f9d5ce5a9dc3a8ca43d .